torch/profiler/profiler.py

```diff
@@ -0,0 +1,283 @@
+import torch.autograd.profiler as prof
+from torch.autograd import ProfilerActivity
+
+from enum import Enum
+from typing import Any, Callable, Iterable, Optional
+from warnings import warn
+
+
+class ProfilerAction(Enum):
+    NONE = 0
+    WARMUP = 1
+    RECORD = 2
+    RECORD_AND_SAVE = 3
+
+
+def schedule(*, wait: int, warmup: int, active: int):
+    """
+    Represents profiler behavior:
+      - wait for 'wait' steps
+      - do the warmup for the next 'warmup' steps
+      - do the active recording for the next 'active' steps
+      - repeat the cycle
+    """
+    def schedule_fn(step: int) -> ProfilerAction:
+        assert step >= 0
+        num_steps = wait + warmup + active
+        mod_step = step % num_steps
+        if mod_step < wait:
+            return ProfilerAction.NONE
+        elif mod_step < wait + warmup:
+            return ProfilerAction.WARMUP
+        else:
+            return ProfilerAction.RECORD if mod_step < num_steps - 1 \
+                else ProfilerAction.RECORD_AND_SAVE
+    assert wait >= 0 and warmup >= 0 and active > 0, \
+        "Invalid profiler schedule arguments"
+    if warmup == 0:
+        warn("Profiler won't be using warmup, this can skew profiler results")
+    return schedule_fn
+
+
+def _default_schedule_fn(_: int) -> ProfilerAction:
+    """
+    Default profiler behavior - immediately start recording the events,
+    keep doing it on every step
+    """
+    return ProfilerAction.RECORD
+
+
+class profile(object):
+    """
+    PyTorch profiler context manager.
+
+    Arguments:
+        activities - list of activity groups (CPU, CUDA);
+        schedule - a callable takes step (int) as a single parameter and returns
```

Comment:
Do we really want to allow people to pass callable here, instead of wait, warmup, active ints? Seems like writing a valid `schedule` would be hard and no one would do it?

Reply:
will point to the user to use schedule()